### PR TITLE
feat: Added support for custom lambda function

### LINKED ISSUE #171: Custom lambda function
## Is your request related to a new offering from AWS?
No

## Is your request related to a problem? Please describe.
Cannot use my custom lambda function code.

## Describe the solution you'd like.
Ability to overwrite default `source_path` in module lambda ([source](https://github.com/terraform-aws-modules/terraform-aws-notify-slack/blob/master/main.tf#L81))

### PR DESCRIPTION
**Pull request by sobi3ch**
## Description
Add ability to inject your custom lambda function. Fixes #171 

## Motivation and Context
The module itself is great but I found myself in the situation where default `notify_slack.py` doesn't work for me and I need to have ability to replace it with my own function.

## Breaking Changes
Nope

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [ ] I have executed `pre-commit run -a` on my pull request

I've install it with `/examples/custom-lambda/` in my aws account and it works.
Before last commit i've run:
```
terraform fmt
terraform validate
```

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -28,6 +28,12 @@ variable "lambda_description" {
   default     = null
 }
 
+variable "lambda_source_path" {
+  description = "The source path of the Lambda function"
```
Comment (antonbabenko) [inline variables.tf:32]:
```suggestion
  description = "The source path of the custom Lambda function"
```

File: main.tf [hunk 2]
```diff
@@ -22,6 +22,8 @@ locals {
     actions   = ["kms:Decrypt"]
     resources = [var.kms_key_arn]
   }
+
+  lambda_handler = split(".", basename(var.lambda_source_path))[0]
```
Comment (antonbabenko) [inline main.tf:26]:
```suggestion
  lambda_handler = try(split(".", basename(var.lambda_source_path))[0], "notify_slack")
```

`var.lambda_source_path` can be `null`

File: examples/custom-lambda/main.tf [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module "notify_slack" {
```
Comment (antonbabenko) [inline examples/custom-lambda/main.tf:1]: Please more this module call into `examples/notify-slack-simple` for brevity, and verify it by running the whole example (with standard and custom source code).

**Review by antonbabenko — CHANGES_REQUESTED** [review]

File: examples/custom-lambda/main.tf (hunk 3 above)
Comment (sobi3ch) [reply examples/custom-lambda/main.tf]: Done. I've moved it to `examples/notify-slack-simple/custom-lambda.tf` and I've test it as you suggested.

**Comment by antonbabenko** [comment]
This PR is included in version 5.2.0 :tada:

**Pull request merged**